.github/workflows/shellcheck.yml

```
@@ -0,0 +1,17 @@
+name: Shellcheck
+on:
+  pull_request:
+    branches:
+      - master
+jobs:
+  shellcheck:
+    name: Shellcheck
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - name: Run ShellCheck
+      uses: ludeeus/action-shellcheck@2.0.0
+      env:
+        SHELLCHECK_OPTS: -x # allow outside sources
+      with:
+        ignore_paths: ./secp256k1-sys/depend/**/*.sh
```

What do you think now, @apoelstra?

---

Does it need a comment? Like
```suggestion
        # this is an upstream depency
        ignore_paths: ./secp256k1-sys/depend/**/*.sh
```

---

I think it's fine. If you need to force-push again it'd probably be good to say
```
# This code comes directly from upstream libsecp256k1 and should not be linted here.
```

---

Perfect. I (and probably we) do have a bad memory and I think that one extra comment line might save some time in the future.